src/KubernetesClient/Kubernetes.ConfigInit.cs

```diff
@@ -257,15 +260,32 @@ private void SetCredentials(KubernetesClientConfiguration config, HttpClientHand
             {
                 chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
 
-                // add all your extra certificate chain
-                chain.ChainPolicy.ExtraStore.Add(caCert);
+                // Added our trusted certificates to the chain
+                //
+                foreach (var caCert in caCerts)
+                {
+                    chain.ChainPolicy.ExtraStore.Add(caCert);
+                }
+
                 chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
                 var isValid = chain.Build((X509Certificate2)certificate);
 
+                var isTrusted = false;
+
                 var rootCert = chain.ChainElements[chain.ChainElements.Count - 1].Certificate;
-                isValid = isValid && rootCert.RawData.SequenceEqual(caCert.RawData);
 
-                return isValid;
+                // Make sure that one of our trusted certs exists in the chain provided by the server.
+                //
+                foreach (var cert in caCerts)
```

Comment:
IMHO, `chain.Build` is enough to validating the server certificate when ca bundles were added. 
I think we can remove the comparison with root cert during this change

Reply:
I kept this around because of [this issue](https://github.com/dotnet/corefx/issues/30284) which was resolved in [this](https://github.com/kubernetes-client/csharp/pull/174) pr. 

From the post it seems like because we are using the flag to not validate the certificate authority that we need to do a manual check to make sure that we trust the certificate chain (based on our CA certs)